functions/spanner/src/main/java/functions/HelloSpanner.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions;
+
+// [START spanner_functions_quickstart]
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import java.io.BufferedWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// HelloSpanner is an example of querying Spanner from a Cloud Function.
+public class HelloSpanner implements HttpFunction {
+  private static final Logger LOGGER = Logger.getLogger(HelloSpanner.class.getName());
+
+  @VisibleForTesting
+  static Spanner createSpanner() {
+    return SpannerOptions.newBuilder().build().getService();
+  }
+
+  // SpannerHolder is a holder class for a Spanner instance that is initialized lazily.
+  private static final class SpannerHolder {
+    private final Object lock = new Object();
+    private volatile boolean initialized;
+    private volatile DatabaseClient client;
+    private volatile Throwable error;
+
+    private SpannerHolder() {}
+
+    // Initialize the {@link Spanner} instance in a method and not as a static variable, as it
+    // might throw an error, and we want to catch and log that specific error.
+    DatabaseClient get() throws Throwable {
```

Comment:
I take it there's no way to simplify this (e.g. with `vavr.io`)?

Reply:
We tend to not bring in a lot of stuff for our samples, that said, when we do, it's usually from Google written & supported libraries like Guava.

In general, we prefer the most specific error when throwing things in samples rather than the least, but this appears appropriate in this instance.

That said, the comment might be extended to discuss what to do if an error is thrown as this object is effectively unusable should that occur.

Reply:
I did look for a standard implementation for this, and the best fit that I could find was `https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/concurrent/LazyInitializer.html`, but I didn't want to introduce an external dependency. That being said, I was surprised that I wasn't able to find anything similar in the Google client libraries, as this is something that I would expect to be a common pattern to use for Cloud Functions. (It could also be that I have overlooked an existing class somewhere, so please feel free to correct me if there is one in the Google libraries.)

Reply:
I've extended the comment to indicate what action should be taken if the initialization fails.